web/models/run.ex

```diff
@@ -5,20 +5,37 @@ defmodule Gatherto.Run do
     field :title, :string
     field :description, :string
     field :time, Ecto.DateTime
+    field :minimum_distance, :integer
+    field :maximum_distance, :integer
     belongs_to :created_by, Gatherto.Athlete, foreign_key: :athlete_id
     many_to_many :athletes, Gatherto.Athlete, join_through: Gatherto.AthleteRun
 
     timestamps()
   end
 
   @required_params [:title, :description, :time]
+  @optional_params [:minimum_distance, :maximum_distance]
 
   @doc """
   Builds a changeset based on the `struct` and `params`.
   """
   def changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, @required_params)
+    |> cast(params, @required_params, @optional_params)
     |> validate_required(@required_params)
+    |> validate_number(:minimum_distance, greater_than: 0)
+    |> validate_number(:maximum_distance, greater_than: 0)
+    |> validate_distance_range
   end
+
+  defp validate_distance_range(changeset) do
+    min_dist = get_field(changeset, :minimum_distance)
+    max_dist = get_field(changeset, :maximum_distance)
+    validate_distance_range(changeset, min_dist, max_dist)
+  end
+
+  defp validate_distance_range(changeset, min_dist, max_dist) when is_integer(min_dist) and min_dist > max_dist do
+    add_error(changeset, :minimum_distance, "minimum distance cannot be more than maximum")
```

Comment:
Just wondering if this is ```is_integer``` on line 37 because it might be nil and if so are there any cases outside of ```nil``` where we'd get in here? If so, we'd want to check max_distance too right? otherwise we'd get:

```elixir
iex(1)> min_dist = 3       
3
iex(2)> max_dist = "2"
"2"
iex(3)> min_dist > max_dist    
false
```